### PR TITLE
chore: fix located_script_name test

### LINKED ISSUE #18417: located_script_name is broken on main
```
---- tests::located_script_name stdout ----
thread 'tests::located_script_name' panicked at 'assertion failed: located_script_name!().starts_with(\"[ext:core/lib.rs:\")', core\lib.rs:168:5
stack backtrace:
```

https://github.com/denoland/deno/actions/runs/4511713940/jobs/7944196384

### PR DESCRIPTION
**Pull request by dsherret**
Closes #18417

### CODE REVIEW

File: .github/workflows/ci.generate.ts [hunk 1]
```diff
@@ -214,6 +213,12 @@ const ci = {
       name: "${{ matrix.job }} ${{ matrix.profile }} ${{ matrix.os }}",
       "runs-on": "${{ matrix.runner || matrix.os }}",
       "timeout-minutes": 120,
+      defaults: {
+        run: {
+          // GH actions doesn't use `set -e` by default unless you specify bash
+          shell: "bash",
```
Comment (dsherret) [inline .github/workflows/ci.generate.ts:219]: It seems like a terrible decision/oversight that they didn't make this the default https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference

**Review by bartlomieju — APPROVED** [review]
LGTM

**Review by mmastrac — APPROVED** [review]
Doh. How did that Windows build pass?

**Comment by dsherret** [comment]
Essentially GH actions does not have `set -e` (fail fast) as the default, which is a bad default IMO: https://stackoverflow.com/questions/62996357/stop-multiline-github-action-on-first-fail

It looks like there are a lot of resources online and other github repos that are doing this wrong too.

**Comment by dsherret** [comment]
Someone followed up with me on the SO question and I realized this only occurs on Windows runners.

**Pull request merged**